src/main.rs

```diff
@@ -33,39 +29,42 @@ fn main() -> anyhow::Result<()> {
 
     let opts = opts::options().run();
 
-    let mut cfg = Config::default()?;
-    cfg.configure(
-        opts.format.verbosity,
-        false,
-        None,
-        opts.frozen,
-        opts.locked,
-        opts.offline,
-        &opts.target_dir,
-        &[],
-        &[],
-    )?;
-
-    let ws = Workspace::new(&opts.manifest_path, &cfg)?;
-    let package = opts::select_package(&opts, &ws);
-    let rustc = cfg.load_global_rustc(Some(&ws))?;
-    let kind = match &opts.target {
-        Some(t) => CompileKind::Target(CompileTarget::new(t)?),
-        None => CompileKind::Host,
+    let cargo_path = std::env::var("CARGO").unwrap_or_else(|_| "cargo".into());
+    let rustc_path = std::env::var("RUSTC").unwrap_or_else(|_| "rustc".into());
+
+    let sysroot = std::process::Command::new(&rustc_path)
```

Comment:
I have a project with a workspace inside `master` folder and doing following steps:

1. change `master/rust-toolchain` file in a project to have 1.63.0 selected
2. `rustup toolchain  uninstall 1.63.0-x86_64-unknown-linux-gnu`
3. `rm -rf master/target`
4. `cargo-asm --manifest-path master/Cargo.toml  -p isin --lib`

With current version this results in it trying to compile stuff, installing 1.63 toolchain and failing to compile because some crates are compiled with 1.64 and some - with 1.63. Probably a cargo (?) bug to be reported...

With `no-libcargo` branch this sequence compiles everything with 1.64, no compilation failures.

Reply:
Interesting. No objections then.